### PR TITLE
Topic 01 exercise 04

### PR DESCRIPTION
**Pull request by fyunes**
Exercise 04,05 & 06

### CODE REVIEW

File: Topic-01-JSExercise-01/script.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    let configObject = {
+        method: 'GET',
+        url: 'https://api.github.com/search/repositories?q=' + searchForm.value,
+    };
+    let promise = fetcher(configObject);
+    promise.then(function (data) {
+        let list = document.createElement('ul');
+        data.items.forEach(function (repo) {
+            let newResult = document.createElement('li');
+            let repoResult = document.createTextNode(repo.id + ': ' + repo.full_name);
```
Comment (DemianMoschin-GLB) [inline Topic-01-JSExercise-01/script.js]: Can you make use of string litereals here?

Comment (DemianMoschin-GLB) [reply Topic-01-JSExercise-01/script.js]: Arrow functions preferred...

File: Topic-01-JSExercise-01/script.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+fetchJoke.onclick = pHandler;
+function funfirst() {
+    hidden.className = 'visible';
+}
+function displayAlert() {
+    alert('Command And Conquer RED ALERT!!!')
+}
+function fetcher(configObject) {
+    let httpfetch = new XMLHttpRequest();
+    let promise = new Promise(function (resolve, reject) {
```
Comment (DemianMoschin-GLB) [inline Topic-01-JSExercise-01/script.js]: Make use of arrow functions here...

File: Topic-01-JSExercise-01/style.css [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    opacity: 0;
+    transition: all ease-in-out 1s;
+}
+.visible {
+    opacity: 1;
+    transition: all ease-in-out 1s;
+}
+.error {
+    background: red;
+}
```
Comment (DemianMoschin-GLB) [inline Topic-01-JSExercise-01/style.css]: Add empty line after

File: Topic-01-JSExercise-01/script.js (hunk 2 above)
Comment (fyunes) [reply Topic-01-JSExercise-01/script.js]: done

File: Topic-01-JSExercise-01/style.css (hunk 3 above)
Comment (fyunes) [reply Topic-01-JSExercise-01/style.css]: done

File: Topic-01-JSExercise-01/script.js (hunk 1 above)
Comment (fyunes) [reply Topic-01-JSExercise-01/script.js]: done

**Pull request merged**